modules/trusted-firmware-m/Kconfig

```diff
@@ -29,21 +29,22 @@ config TFM_MINIMAL
 	  needs a different TF-M configuration you have to disable this option
 	  and reproduce the desired configuration through kconfig fragments.
 
+source "$(ZEPHYR_BASE)/modules/trusted-firmware-m/Kconfig.tfm"
+
 if !TFM_MINIMAL
 rsource "Kconfig.mbedtls.defconfig"
-source "$(ZEPHYR_BASE)/modules/trusted-firmware-m/Kconfig"
+
+# When enabling full TFM support, then partitions and crypto modules shall be available.
+source "$(ZEPHYR_BASE)/modules/trusted-firmware-m/Kconfig.tfm.partitions"
+source "$(ZEPHYR_BASE)/modules/trusted-firmware-m/Kconfig.tfm.crypto_modules"
```

Comment:
I don't get it.

Assuming Kconfig.tfm.partitions was sourced before this commit then we are now double-sourcing Kconfig.tfm.partitions.

Cut off one head and two will replace it.

Reply:
I'm a bit confused.

if A
source B
endif

does not affect whether B is sourced or not, right?

It only applies a "depends on", on all of B.

Reply:
resolved offline